prime-angular-frontend/src/app/modules/enrolment/pages/self-declaration/self-declaration.component.ts

```diff
@@ -100,9 +102,15 @@ export class SelfDeclarationComponent extends BaseEnrolmentProfilePage implement
   }
 
   public onSubmit() {
-    const hasBeenThroughTheWizard = true;
-    this.hasAttemptedFormSubmission = true;
-    super.onSubmit(hasBeenThroughTheWizard);
+    if (!this.clickedDownloadDoc) {
```

Comment:
What is this code doing? 
It looks to be that I am required to hit the submit button twice if i decided to download a document?
Is this attempting to stop the user from submitting when an action is pending?

Reply:
So played around testing it and I see why this is there. So in this case we need to update the app-document-attachment and make sure that button is updated so it isn't assumed to be a submit button. This has probably never came us since this component isn't usually used in a form.

"Just add type='button' to it. A button without any defined type in a form acts like a submit button for the form."

Reply:
Perfect!